### PR TITLE
check for all allowable iidtoken scopes, not just *

### PR DESCRIPTION
**Pull request by fredquintana**
The code was hardcoded to only check for the wildcard scope when looking for the iidtoken when migrating an iid to a fid.

Since scopes FCM and GCM are also allowed, check for those, too.

### CODE REVIEW

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java [hunk 1]
```diff
@@ -44,9 +44,9 @@
   private static final String STORE_KEY_PUB = "|S||P|";
   private static final String STORE_KEY_ID = "|S|id";
   private static final String STORE_KEY_TOKEN = "|T|";
-  private static final String DEFAULT_SCOPE = "|*";
   private static final String JSON_TOKEN_KEY = "token";
   private static final String JSON_ENCODED_PREFIX = "{";
+  private static final String [] ALLOWABLE_SCOPES = new String[]{"*", "FCM", "GCM"};
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: Is the space on purpose between String and []?

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java [hunk 2]
```diff
@@ -86,14 +86,23 @@ private static String getDefaultSenderId(FirebaseApp app) {
     return projectNumber;
   }
 
-  private String createTokenKey(@NonNull String senderId) {
-    return STORE_KEY_TOKEN + senderId + DEFAULT_SCOPE;
+  private String createTokenKey(@NonNull String senderId, @NonNull String scope) {
+    return STORE_KEY_TOKEN + senderId + " | " + scope;
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: constant for the pipe character "|"

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java [hunk 3]
```diff
@@ -86,14 +86,23 @@ private static String getDefaultSenderId(FirebaseApp app) {
...
   }
 
   @Nullable
   public String readToken() {
     synchronized (iidPrefs) {
-      String token = iidPrefs.getString(createTokenKey(defaultSenderId), null);
+      Log.d("Rayo", "readToken: defaultSenderId: " + defaultSenderId);
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: I know you would have removed it, but still annoying you with a comment, haha

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java [hunk 4]
```diff
@@ -86,14 +86,23 @@ private static String getDefaultSenderId(FirebaseApp app) {
...
+      String token = null;
+
+      for (String scope : ALLOWABLE_SCOPES) {
+        token = iidPrefs.getString(createTokenKey(defaultSenderId, scope), null);
+        if (!TextUtils.isEmpty(token)) {
+          break;
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: How about we replace `break;` by this line:
```
return (token.startsWith(JSON_ENCODED_PREFIX)) ? parseIidTokenFromJson(token) : token;
```


And below instead of
```
if (TextUtils.isEmpty(token)) {
  return null;
}
```
We just do `return null;`

Comment (fredquintana) [reply firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: done

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java (hunk 3 above)
Comment (fredquintana) [reply firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: done

**Comment by fredquintana** [comment]
I also added unit tests for the different supported scopes as well as the empty scope and the json formatted tokens.

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java [hunk 5]
```diff
@@ -44,9 +45,9 @@
   private static final String STORE_KEY_PUB = "|S||P|";
   private static final String STORE_KEY_ID = "|S|id";
   private static final String STORE_KEY_TOKEN = "|T|";
-  private static final String DEFAULT_SCOPE = "|*";
   private static final String JSON_TOKEN_KEY = "token";
   private static final String JSON_ENCODED_PREFIX = "{";
+  private static final String [] ALLOWABLE_SCOPES = new String[]{"*", "FCM", "GCM", ""};
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: Is the space on purpose between String and []?

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java [hunk 6]
```diff
@@ -86,23 +93,22 @@ private static String getDefaultSenderId(FirebaseApp app) {
     return projectNumber;
   }
 
-  private String createTokenKey(@NonNull String senderId) {
-    return STORE_KEY_TOKEN + senderId + DEFAULT_SCOPE;
+  private String createTokenKey(@NonNull String senderId, @NonNull String scope) {
+    return STORE_KEY_TOKEN + senderId + "|" + scope;
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: Maybe a constant for the pipe "|"?

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java (hunk 2 above)
Comment (fredquintana) [reply firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: done

File: firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java (hunk 1 above)
Comment (fredquintana) [reply firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java]: no, done
